### PR TITLE
chore: restore bot workflow to update attributions

### PR DESCRIPTION
**Pull request by cryptodev-2s**
## **Description**

This PR reinstates the workflow that was removed in [this PR](https://github.com/MetaMask/metamask-mobile/pull/9920). Additionally, a safeguard has been added to the workflow to prevent it from running on a fork.

## **Related issues**

Fixes: [#1550](https://github.com/MetaMask/mobile-planning/issues/1550)

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .github/workflows/update-attributions.yml [hunk 1]
```diff
@@ -0,0 +1,192 @@
+name: Update Attibutions
```
Comment (mcmire) [inline .github/workflows/update-attributions.yml:1]: Nit: typo

``` suggestion
name: Update Attributions

File: .github/workflows/update-attributions.yml [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.issue.number }}
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+      - name: Install dependencies from cache
+        run: yarn --immutable --immutable-cache
+      - name: Generate Atributions
```
Comment (mcmire) [inline .github/workflows/update-attributions.yml:95]: Nit: typo

``` suggestion
      - name: Generate Attributions

File: .github/workflows/update-attributions.yml [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        env:
+          HAS_CHANGES: ${{ steps.attributions-changes.outputs.HAS_CHANGES }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.issue.number }}
+
+  check-status:
+    name: Check whether the attributions update succeeded
+    runs-on: ubuntu-latest
+    needs:
+      - commit-updated-attributions
```
Comment (mcmire) [inline .github/workflows/update-attributions.yml:164]: I see that we check for whether the repo is a fork in all other jobs but this one. Do we want to do that here too?

**Review by mcmire — COMMENTED** [review]
A few suggestions, but looks good to me regardless.

**Review by mcmire — DISMISSED** [review]

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**